platform_channels/lib/src/pet_list_message_channel.dart

```diff
@@ -69,15 +69,15 @@ class PetDetails {
     this.breed,
   });
 
-  final String petType;
-  final String breed;
+  final String? petType;
```

Comment:
IIRC, this object is meant to have both fields set, right? Consider making these non-nullable, and updating the fromMap constructor to throw an exception if it can't find the values for them.

Reply:
Sure @RedBrogdon  :)